### ISSUE TITLE
Qwen3-32B Non-deterministic demo output

### ISSUE THREAD

**Issue by djordje-tt** [tenstorrent/tt-metal#38645]
### Problem description
After fixing perf regression and couple other issues. Demo for Qwen3-32B starts showing non-determinism. 
While examining expected outputs I encountered it contains list of expected outputs so it was probably even earlier showing non-determinism which had workaround with list of different outputs which would pass then. Apart from that test contains `repeat2` which means it runs demo twice against same input but looking into the outputs it produces 2 different outputs. Based on my initial investigation first 20ish tokens are the same and it should not be sampling since it's behaving same with top_p=0 and temp=0 (argmax).
### Next steps
Compare across different runs: 
- kv cache after prefill
- logits after prefill
- logits in iter N+1 where tokens are same in iter N and different in iter N+1 in 2 runs
- investigate sampling.

If kv cache or logits after prefill are not identical then bug is in prefill. If logits are not identical in any iter of decode while sampled tokens are the same, divergence starts happening in decode. If logits are identical in all and in iter N+1 then sampling is root cause and should be further investigated.

### Repro command 
```
command: 
HF_MODEL=Q/proj_sw/user_dev/wen3-32B pytest models/demos/llama3_70b_galaxy/demo/text_qwen_demo.py -k repeat2

vscode (launch.json)
{
      "name": "Pytest: qwen demo repeat2",
      "type": "python",
      "request": "launch",
      "module": "pytest",
      "cwd": "${workspaceFolder}/tt-metal",
      "args": [
        "models/demos/llama3_70b_galaxy/demo/text_qwen_demo.py",
        "-k",
        "repeat2"
      ],
      "console": "integratedTerminal",
      "justMyCode": true,
      "env": {
        "TT_METAL_HOME": "/localdev/divanovic/tt-metal",
        "TT_METAL_ENV": "dev",
        "TT_METAL_CACHE": "/localdev/divanovic/.cache",
        "HF_MODEL": "/proj_sw/user_dev/Qwen3-32B",
        "TT_CACHE_PATH": "/localdev/divanovic/cache/qwen3-32b",
        "MESH_DEVICE": "TG",
        "PYTHONPATH": "/localdev/divanovic/tt-metal"
      }
    },```

**Comment by mtairum** [comment]
Issue is tied to https://github.com/tenstorrent/tt-metal/issues/37397. See that issue for latest progress.

**Comment by mtairum** [comment]
@ifabijanic-tt to retest the determinism of qwen3-32B now that some fixes were merged.
It needs to have the same coverage as Llama-3.3-70B and showcase the same behaviour or better.

**Comment by ifabijanic-tt** [comment]
This failure was a part of larger sampling issues on Llama 70B on Galaxy machine which also affected the Qwen model as well because of code shared between the models. PRs for this issue are these (both on tt-metal and vllm side):
https://github.com/tenstorrent/tt-metal/pull/50688
https://github.com/tenstorrent/tt-metal/pull/50687
https://github.com/tenstorrent/tt-metal/pull/50685
https://github.com/tenstorrent/vllm/pull/444

All PRs are merged so this issue shouldn't happen as of today. Closing this issue. If anything new comes up, please open new issue.
